pkg/operator/kubeclient.go

```diff
@@ -29,3 +31,21 @@ func (c *vSphereProblemDetectorController) ListStorageClasses(ctx context.Contex
 func (c *vSphereProblemDetectorController) ListPVs(ctx context.Context) ([]*v1.PersistentVolume, error) {
 	return c.pvLister.List(labels.Everything())
 }
+
+func (c *vSphereProblemDetectorController) GetClusterConfig(ctx context.Context) (*v1.ConfigMap, error) {
+	return c.kubeSystemConfigMapLister.ConfigMaps(kubeSystemNamespace).Get(clusterConfigName)
+}
+
+func (c *vSphereProblemDetectorController) GetUserName(ctx context.Context, cfg *vsphere.VSphereConfig) (*string, error) {
```

Comment:
Can you use or refactor `getVSphereConfig` / `parseConfig` / `getCredentials` from `vmware.go`? I think at least something could be shared.

Reply:
Yes, will take a look at that.